### PR TITLE
Updates design of index pages

### PR DESCRIPTION
**Pull request by mperrotti**
Based on this Figma mock:
https://www.figma.com/file/byiGhUWH17SLpeQStmULYA/Doctocat-for-Primer%3A--Q4-IA-%2B-UX-proposal?type=design&node-id=5-7484&mode=design&t=JV78JYmjJkRbLB2f-4

Before:
Screenshot 2023-07-25 at 5 02 34 PM

After:
Screenshot 2023-07-25 at 4 59 57 PM

### CODE REVIEW

File: content/github-staff/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: GitHub staff
+description: Internal documentation for GitHub staff.
+---
+
+<ChildPages of="GitHub staff">
```
Comment (josepmartins) [inline content/github-staff/index.md:6]: @mperrotti This is probably causing the issues on the build ❤️ 

```suggestion
<ChildPages of="GitHub staff" />
```

**Review by emilybrick — APPROVED** [review]
❇️

File: src/components/child-pages.tsx [hunk 2]
```diff
@@ -49,12 +49,12 @@ export function ChildPages({of: title}: {of: string}) {
     <Box sx={{display: 'grid', gap: 5}}>
       {childItems.map(item => (
         <Box key={item.url} sx={{display: 'grid', gap: 1}}>
-          <Heading as="h3" sx={{fontSize: 3}}>
+          <Heading as="h3" sx={{fontSize: 4}}>
```
Comment (emilybrick) [inline src/components/child-pages.tsx:52]:
```suggestion
          <Heading as="h3" sx={{fontSize: 3}}>
```

File: src/components/child-pages.tsx [hunk 3]
```diff
@@ -49,12 +49,12 @@ export function ChildPages({of: title}: {of: string}) {
...
             <Link as={GatsbyLink} to={item.url}>
               {item.title}
             </Link>
           </Heading>
-          <Text as="p" sx={{m: 0, maxWidth: '80ch'}}>
+          <Text as="p" sx={{m: 0, maxWidth: '80ch', fontSize: 3, color: 'fg.subtle'}}>
```
Comment (emilybrick) [inline src/components/child-pages.tsx:57]:
```suggestion
          <Text as="p" sx={{m: 0, maxWidth: '80ch', fontSize: 2, color: 'fg.subtle'}}>
```

File: src/layouts/index-layout.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export default function IndexLayout(props) {
+  const {pageContext, children} = props
+  const {title, description} = pageContext.frontmatter
+
+  return (
+    <BaseLayout title={title} description={description}>
+      <Box sx={{maxWidth: 1200, width: '100%', p: [4, 5, 6, 7], mx: 'auto'}}>
+        <Box sx={{display: "flex", flexDirection: 'column', gap: 2, mb: 7}}>
+          {/* TODO: update `fontSize` to `8` once we can upgrade to the latest version of PRC, which supports that fontSize value */}
+          <Heading as="h1" sx={{fontSize: 56}}>{title}</Heading>
```
Comment (emilybrick) [inline src/layouts/index-layout.tsx:17]:
```suggestion
          <Heading as="h1" sx={{fontSize: 48}}>{title}</Heading>
```

**Review by josepmartins — APPROVED** [review]

**Pull request merged**